deploy/runtools/run_farm.py

```diff
@@ -1,361 +1,397 @@
+        available_instances_per_handle = {}
+        for sim_host_handle in self.SIM_HOST_HANDLE_TO_MAX_FPGA_SLOTS:
+            available_instances_per_handle[sim_host_handle] = instances_sorted_by_avail_ip(get_run_instances_by_tag_type(self.run_farm_tag, sim_host_handle))
 
         message = """Insufficient {}. Did you run `firesim launchrunfarm`?"""
         # confirm that we have the correct number of instances
-        if not (len(available_f1_16_instances) >= len(self.f1_16s)):
-            rootLogger.warning(message.format("f1.16xlarges"))
-        if not (len(available_f1_4_instances) >= len(self.f1_4s)):
-            rootLogger.warning(message.format("f1.4xlarges"))
-        if not (len(available_f1_2_instances) >= len(self.f1_2s)):
-            rootLogger.warning(message.format("f1.2xlarges"))
-        if not (len(available_m4_16_instances) >= len(self.m4_16s)):
-            rootLogger.warning(message.format("m4.16xlarges"))
+        for sim_host_handle in self.SIM_HOST_HANDLE_TO_MAX_FPGA_SLOTS:
+            if not (len(available_instances_per_handle[sim_host_handle]) >= len(self.run_farm_hosts_dict[sim_host_handle])):
+                rootLogger.warning(message.format(sim_host_handle))
 
         ipmessage = """Using {} instances with IPs:\n{}"""
-        rootLogger.debug(ipmessage.format("f1.16xlarge", str(get_private_ips_for_instances(available_f1_16_instances))))
-        rootLogger.debug(ipmessage.format("f1.4xlarge", str(get_private_ips_for_instances(available_f1_4_instances))))
-        rootLogger.debug(ipmessage.format("f1.2xlarge", str(get_private_ips_for_instances(available_f1_2_instances))))
-        rootLogger.debug(ipmessage.format("m4.16xlarge", str(get_private_ips_for_instances(available_m4_16_instances))))
+        for sim_host_handle in self.SIM_HOST_HANDLE_TO_MAX_FPGA_SLOTS:
+            rootLogger.debug(ipmessage.format(sim_host_handle, str(get_private_ips_for_instances(available_instances_per_handle[sim_host_handle]))))
 
         # assign boto3 instance objects to our instance objects
-        for index, instance in enumerate(available_f1_16_instances):
-            self.f1_16s[index].assign_boto3_instance_object(instance)
-
-        for index, instance in enumerate(available_f1_4_instances):
-            self.f1_4s[index].assign_boto3_instance_object(instance)
-
-        for index, instance in enumerate(available_m4_16_instances):
-            self.m4_16s[index].assign_boto3_instance_object(instance)
-
-        for index, instance in enumerate(available_f1_2_instances):
-            self.f1_2s[index].assign_boto3_instance_object(instance)
-
+        for sim_host_handle in self.SIM_HOST_HANDLE_TO_MAX_FPGA_SLOTS:
+            for index, instance in enumerate(available_instances_per_handle[sim_host_handle]):
+                inst_tup = self.run_farm_hosts_dict[sim_host_handle][index]
+                inst = inst_tup[0]
+                inst.set_host("centos@" + instance.private_ip_address)
```

Comment:
is adding the `centos@` prefix here really required?  Have we always done this? I don't see it in the old code and I'm concerned that it will break `RunFarm.lookup_by_host()`.

Reply:
See other comment - Basically no. No this wasn't done before, I'll revert this. This doesn't break the lookup by host though.